src/tools/distillery.ml

```diff
@@ -194,6 +195,32 @@ let get_templates () =
       | End_of_file -> Unix.closedir dir; rl
   in aux []
 
+(* Check if the project name is valid for the given template. For example,
+ * options is not a valid project name for basic.ppx. It is supposed each
+ * template has a file {!forbidden_project_name_filename} file containing the
+ * list of forbidden project name (one name a line).
+ *)
+let check_forbidden_project_name project_name template_name =
+  let file = Filename.concat
+    (Filename.concat (get_templatedir ()) template_name)
+    forbidden_project_name_filename
+  in
+  if Sys.file_exists file
+  then
+  (
+    let in_file = open_in file in
+    try
+      let rec locale () =
```

Comment:
The name is confusing to non-french speakers; locale in computing generally points to internationalization :). Maybe `aux` or just `f`?